hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapreduce/lib/input/TestFileInputFormat.java

```diff
@@ -238,6 +238,44 @@ public void testListStatusErrorOnNonExistantDir() throws IOException {
     }
   }
 
+  @Test
+  public void testShrinkStatus() throws IOException {
+    Configuration conf = getConfiguration();
+    MockFileSystem mockFs =
+            (MockFileSystem) new Path("test:///").getFileSystem(conf);
+    Path dir1  = new Path("test:/a1");
+    RemoteIterator<LocatedFileStatus> statuses = mockFs.listLocatedStatus(dir1);
```

Comment:
This test doesn't test what we care about in this change.  `MockFileSystem` doesn't return a derived class of `BlockLocation` so nothing shrinks.  This test passes if `shrinkStatus` simply returned the original status, showing it really isn't a useful test for verifying we don't regress and start caching large location statuses again.

Reply:
The updated codes have made ```MockFileSystem.getFileBlockLocations``` return array of ```HdfsBlockLocation``` for regression and the unit test shows that the filestatus being shrunk has the same block location infos with the original.

Reply:
Thanks for the updates!   The unit test still has the problem I originally described -- it will pass even when the shrink implementation does not actually shrink anything.  Maybe the unit test should assert that the `LocatedFileStatus` received from the filesystem is using instances of `HdfsBlockLocation` (which it does after your recent changes) and the result of shrinking no longer has such instances?

Reply:
Thank you for your reply! Add assert here to check the instance of original filestatus used and the result of shrinking should be a ```BlockLocation```,  which contains no references to ```LocatedBlock```.